setup.py

```diff
@@ -1,16 +1,47 @@
 import os
 from setuptools import setup, Extension
-from Cython.Build import cythonize
-import numpy
+from distutils.command.sdist import sdist as _sdist
+
+try:
+    import numpy
+except ImportError:
+    use_numpy = False
+else:
+    use_numpy = True
+
+try:
+    from Cython.Build import cythonize
+except ImportError:
+    use_cython = False
+else:
+    use_cython = True
 
 name = "ndsplines"
 
-extensions = [
-    Extension(f"{name}._bspl",
-              [os.path.join(name, "_bspl.pyx")],
-              include_dirs=[numpy.get_include()],
-              depends=[os.path.join(name, "_bspl.h")]),
-]
+cmdclass = {}
+
+if use_cython and use_numpy:
+    extensions = cythonize([
+        Extension(f"{name}._bspl",
```

Comment:
Should we make an `extension = "_bspl"` variable and if you're cool with requiring Python >= 3.6, use f strings like here to replace all occurrences of `_bspl` with `{extension}`.

Reply:
A lot of projects are moving on to require > 3.4, but I think 3.5 support is still pretty widespread. For me, f strings are nice but not really worth being the sole reason for a Python version requirement.

Reply:
I neither feel motivated to support <= 3.5 nor do I feel strongly about f-strings over some other way to DRY up references to the name.  I guess we should be consistent though? If you don't mind I'll defer to you on this. I believe the f-string usage here came from your commit.